src/commands/init.ts

```diff
@@ -96,30 +95,57 @@ function getProjectConfig(type) {
         ],
       }
     default:
-      return {} // TODO: Probably throw
+      throw new Error()
   }
 }
 
-async function actionHandler(projectType, projectName, options) {
+type InitOptions = {
+  openCode?: boolean
+}
+
+async function actionHandler(
+  projectType?: ProjectType,
+  projectName?: string,
+  options?: InitOptions
+): Promise<void> {
   let isCancel = false
   const onCancel = () => (isCancel = true)
 
+  const promptQuestions: PromptObject[] = []
+  if (!projectType) {
+    promptQuestions.push({
+      type: 'select',
+      name: 'projectType',
+      message: 'Which type of project would you like to create?',
+      choices: [
+        ...webProjectTypes,
+        {
+          value: ProjectType.ReactNative,
+          title: 'React Native',
+        },
+      ],
+    })
+  }
+  if (!projectName) {
+    promptQuestions.push({
+      type: isCancel ? null : 'text',
```

Comment:
Appears I should change this back to individual `await` prompts so the `isCancel` flag gets evaluated at runtime before prompting with the next question.